src/bootstrap/config.rs

```diff
@@ -857,7 +857,7 @@ impl Config {
         let build = toml.build.unwrap_or_default();
 
         set(&mut config.initial_rustc, build.rustc.map(PathBuf::from));
-        set(&mut config.out, build.build_dir.map(PathBuf::from));
+        set(&mut config.out, flags.build_dir.or_else(|| build.build_dir.map(PathBuf::from)));
```

Comment:
nit: doesn't matter now, but in general bootstrap prefers the simple way over the performant way to do things; we care about compile time more than runtime.
```suggestion
        set(&mut config.out, flags.build_dir.or(build.build_dir.map(PathBuf::from));
```

Reply:
Ah, I wasn't sure if you ran clippy on it or something (since that's a lint on by default in clippy). I doubt the compile time of these is measurably different, FWIW.

Reply:
(That said I don't mind adding the change, but you've already r+ed it, so...)